src/netex_idf/calendars.rs

```diff
@@ -41,19 +41,43 @@ struct ValidityPatternIterator<'a> {
     current_date: Date,
     end_date: Date,
     weekday_pattern: &'a HashSet<Weekday>,
-    day_type_assignments: Option<&'a Vec<DayTypeAssignment<'a>>>,
+    day_type_assignments: Vec<&'a DayTypeAssignment<'a>>,
 }
 impl<'a> ValidityPatternIterator<'a> {
     fn new(
         validity_period: &ValidityPeriod,
         weekday_pattern: &'a HashSet<Weekday>,
         day_type_assignments: Option<&'a Vec<DayTypeAssignment<'a>>>,
     ) -> ValidityPatternIterator<'a> {
+        let mut dtas = vec![];
+        let mut ops = vec![];
+        let mut actives = vec![];
+        let mut inactives = vec![];
+
+        if let Some(dtas) = day_type_assignments {
+            for dta in dtas {
+                match dta {
+                    DayTypeAssignment::OperatingPeriod(_) => ops.push(dta),
+                    DayTypeAssignment::ActiveDay(_) => actives.push(dta),
+                    DayTypeAssignment::InactiveDay(_) => inactives.push(dta),
+                }
+            }
+        }
+
+        // operating periods are treated first
+        // then active days
+        // then inactive days
+        // to ensure that inactive days are removed if they appears before
+        // operating periods in the XML feed.
+        dtas.extend(ops);
+        dtas.extend(actives);
+        dtas.extend(inactives);
```

Comment:
What is done here is basically ordering objects of type `DayTypeAssignment`. How about implementing `PartialOrd` and `Ord` instead? I gave it a try in [this PR](https://github.com/datanel/transit_model/pull/3) and it works pretty well :smile:

Reply:
it seems complicated but more idiomatic